src/marray.spec.ts

```diff
@@ -373,6 +373,21 @@ describe('array', () => {
 
       expect(result1).toEqual(result2);
     });
+
+    it('should behave in a same way as a regular array when being iterated with a MArray.forEach', () => {
+      const regArray = [2, 4, 6, 8, 10];
+      const mArray = new MArray(...regArray);
+      const regResult: [number, number, number[]][] = [];
+      const marrayResult: typeof regResult = [];
+
+      regArray.forEach((v, i, a) => regResult.push([v, i, a]));
+      mArray.forEach((v, i, a) => {
+        expect(a).toBeInstanceOf(MArray);
```

Comment:
I think the expected type checks for this :D

Reply:
Changed with equality check (`a===mArray`) in 62498a7